[Bug][Regression] This project failed to configure when project name contains space

### Brief Issue Summary

#### Environment:
VS Code 1.99.2+ CMake Tools v1.21.15 (pre-release)

#### More info:
1. Repro on Windows, Linux, Mac
2. It doesn't repro on v1.21.13 and starts repro from v1.21.14 to latest v1.21.15, so this is a regression issue

#### Repro steps:
1. Create a folder named **demo project** (**folder name contains space**) and open it with VS Code
2. Create project using command `CMake: Quick Start`
3. Run command `CMake: Configure`

#### Expected Result:
The project configures successfully

#### Actual Result:
This project failed to configure

> [proc] Executing command: "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" --version
[proc] Executing command: "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -E capabilities
[kit] Successfully loaded 19 kits from C:\Users\v-amyli3\AppData\Local\CMakeTools\cmake-tools-kits.json
[presetController] Reading and validating the presets file "C:\Users\v-amyli3\Desktop\Bi-weekly-app\demo project\CMakePresets.json"
[presetController] Successfully validated C:\Users\v-amyli3\Desktop\Bi-weekly-app\demo project\CMakePresets.json against presets schema
[presetController] Expanding presets file C:\Users\v-amyli3\Desktop\Bi-weekly-app\demo project\CMakePresets.json
[presetController] Successfully expanded presets file C:\Users\v-amyli3\Desktop\Bi-weekly-app\demo project\CMakePresets.json
[main] Configuring project: demo project 
[proc] Executing command: "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -DCMAKE_BUILD_TYPE=Debug "-DCMAKE_INSTALL_PREFIX=C:/Users/v-amyli3/Desktop/Bi-weekly-app/demo project/out/install/MyPresets" -S C:/Users/v-amyli3/Desktop/Bi-weekly-app/demo%20project -B C:/Users/v-amyli3/Desktop/Bi-weekly-app/demo%20project/out/build/MyPresets -G Ninja
[cmake] **CMake Error: The source directory "C:/Users/v-amyli3/Desktop/Bi-weekly-app/demo%20project" does not exist.**
[cmake] Specify --help for usage, or press the help button on the CMake GUI.
[proc] The command: "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -DCMAKE_BUILD_TYPE=Debug "-DCMAKE_INSTALL_PREFIX=C:/Users/v-amyli3/Desktop/Bi-weekly-app/demo project/out/install/MyPresets" -S C:/Users/v-amyli3/Desktop/Bi-weekly-app/demo%20project -B C:/Users/v-amyli3/Desktop/Bi-weekly-app/demo%20project/out/build/MyPresets -G Ninja exited with code: 1

![Image](https://github.com/user-attachments/assets/154ea30e-8f75-4481-ada4-a9729410efb2)

### CMake Tools Diagnostics

```shell
{
  "os": "win32",
  "vscodeVersion": "1.99.2",
  "cmtVersion": "1.21.15",
  "configurations": [
    {
      "folder": "c:\\Users\\v-amyli3\\Desktop\\Bi-weekly-app\\demo project",
      "cmakeVersion": "3.30.5",
      "configured": true,
      "generator": "Ninja",
      "usesPresets": true,
      "compilers": {
        "C": "C:/msys64/ucrt64/bin/cc.exe",
        "CXX": "C:/msys64/ucrt64/bin/c++.exe"
      }
    }
  ],
  "cpptoolsIntegration": {
    "isReady": false,
    "hasCodeModel": false,
    "activeBuildType": "",
    "buildTypesSeen": [],
    "requests": [],
    "responses": [],
    "partialMatches": [],
    "targetCount": 0,
    "executablesCount": 0,
    "librariesCount": 0,
    "targets": []
  },
  "settings": [
    {
      "communicationMode": "automatic",
      "useCMakePresets": "auto",
      "configureOnOpen": true
    }
  ]
}
```

### Debug Log

```shell
[main] Configuring project: demo project 
[main] Saving open files before configure/build
[driver] Start configure 
[driver] Running pre-configure checks and steps
[cmakefileapi-driver] Configuring using preset
[cmakefileapi-driver] Invoking CMake C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe with arguments ["-DCMAKE_BUILD_TYPE=Debug","-DCMAKE_INSTALL_PREFIX=C:/Users/v-amyli3/Desktop/Bi-weekly-app/demo project/out/install/MyPresets","-S","C:/Users/v-amyli3/Desktop/Bi-weekly-app/demo%20project","-B","C:/Users/v-amyli3/Desktop/Bi-weekly-app/demo%20project/out/build/MyPresets","-G","Ninja"]
[proc] Executing command: "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -DCMAKE_BUILD_TYPE=Debug "-DCMAKE_INSTALL_PREFIX=C:/Users/v-amyli3/Desktop/Bi-weekly-app/demo project/out/install/MyPresets" -S C:/Users/v-amyli3/Desktop/Bi-weekly-app/demo%20project -B C:/Users/v-amyli3/Desktop/Bi-weekly-app/demo%20project/out/build/MyPresets -G Ninja
[proc]   with environment: {"ALLUSERSPROFILE":"C:\\ProgramData","APPDATA":"C:\\Users\\v-amyli3\\AppData\\Roaming","CHROME_CRASHPAD_PIPE_NAME":"\\\\.\\pipe\\crashpad_9928_WWCTANHDOGUXRZLM","CLIENTNAME":"DESKTOP-4AMU057","COMPUTERNAME":"CPC-vamyl-W9UOR","ChocolateyInstall":"C:\\ProgramData\\chocolatey","ChocolateyLastPathUpdate":"133823496213910293","ComSpec":"C:\\Windows\\system32\\cmd.exe","CommonProgramFiles":"C:\\Program Files\\Common Files","CommonProgramFiles(x86)":"C:\\Program Files (x86)\\Common Files","CommonProgramW6432":"C:\\Program Files\\Common Files","DriverData":"C:\\Windows\\System32\\Drivers\\DriverData","EFC_12152":"1","ELECTRON_RUN_AS_NODE":"1","FPS_BROWSER_APP_PROFILE_STRING":"Internet Explorer","FPS_BROWSER_USER_PROFILE_STRING":"Default","HOMEDRIVE":"C:","HOMEPATH":"\\Users\\v-amyli3","IsDevBox":"True","LOCALAPPDATA":"C:\\Users\\v-amyli3\\AppData\\Local","LOGONSERVER":"\\\\CPC-vamyl-W9UOR","NUMBER_OF_PROCESSORS":"16","ORIGINAL_XDG_CURRENT_DESKTOP":"undefined","OS":"Windows_NT","OneDrive":"C:\\Users\\v-amyli3\\OneDrive - Microsoft","OneDriveCommercial":"C:\\Users\\v-amyli3\\OneDrive - Microsoft","PATHEXT":".COM;.EXE;.BAT;.CMD;.VBS;.VBE;.JS;.JSE;.WSF;.WSH;.MSC","PROCESSOR_ARCHITECTURE":"AMD64","PROCESSOR_IDENTIFIER":"AMD64 Family 25 Model 1 Stepping 1, AuthenticAMD","PROCESSOR_LEVEL":"25","PROCESSOR_REVISION":"0101","PSModulePath":"C:\\Program Files\\WindowsPowerShell\\Modules;C:\\Windows\\system32\\WindowsPowerShell\\v1.0\\Modules","PUBLIC":"C:\\Users\\Public","Path":"C:\\Windows\\system32;C:\\Windows;C:\\Windows\\System32\\Wbem;C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\;C:\\Windows\\System32\\OpenSSH\\;C:\\Windows\\system32\\config\\systemprofile\\AppData\\Local\\Microsoft\\WindowsApps;C:\\Program Files\\Microsoft Dev Box Agent\\Scripts;C:\\Program Files\\Microsoft SQL Server\\150\\Tools\\Binn\\;C:\\Program Files (x86)\\Windows Kits\\10\\Windows Performance Toolkit\\;C:\\msys64\\ucrt64\\bin;C:\\ProgramData\\chocolatey\\bin;C:\\msys64\\usr\\bin;C:\\Program Files\\Git\\cmd;C:\\Users\\v-amyli3\\AppData\\Local\\Microsoft\\WindowsApps;C:\\Users\\v-amyli3\\AppData\\Local\\Programs\\Microsoft VS Code\\bin","ProgramData":"C:\\ProgramData","ProgramFiles":"C:\\Program Files","ProgramFiles(x86)":"C:\\Program Files (x86)","ProgramW6432":"C:\\Program Files","SESSIONNAME":"rdp-sxs250105600#0","SystemDrive":"C:","SystemRoot":"C:\\Windows","TEMP":"C:\\Users\\v-amyli3\\AppData\\Local\\Temp","TMP":"C:\\Users\\v-amyli3\\AppData\\Local\\Temp","UATDATA":"C:\\Windows\\CCM\\UATData\\D9F8C395-CAB8-491d-B8AC-179A1FE1BE77","USERDNSDOMAIN":"redmond.corp.microsoft.com","USERDOMAIN":"REDMOND","USERDOMAIN_ROAMINGPROFILE":"REDMOND","USERNAME":"v-amyli3","USERPROFILE":"C:\\Users\\v-amyli3","VSCODE_CODE_CACHE_PATH":"C:\\Users\\v-amyli3\\AppData\\Roaming\\Code\\CachedData\\4949701c880d4bdb949e3c0e6b400288da7f474b","VSCODE_CRASH_REPORTER_PROCESS_TYPE":"extensionHost","VSCODE_CWD":"C:\\Users\\v-amyli3\\AppData\\Local\\Programs\\Microsoft VS Code","VSCODE_ESM_ENTRYPOINT":"vs/workbench/api/node/extensionHostProcess","VSCODE_HANDLES_UNCAUGHT_ERRORS":"true","VSCODE_IPC_HOOK":"\\\\.\\pipe\\67ed9be6-1.99.2-main-sock","VSCODE_L10N_BUNDLE_LOCATION":"","VSCODE_NLS_CONFIG":"{\"userLocale\":\"en\",\"osLocale\":\"en-us\",\"resolvedLanguage\":\"en\",\"defaultMessagesFile\":\"C:\\\\Users\\\\v-amyli3\\\\AppData\\\\Local\\\\Programs\\\\Microsoft VS Code\\\\resources\\\\app\\\\out\\\\nls.messages.json\",\"locale\":\"en\",\"availableLanguages\":{}}","VSCODE_PID":"9928","WVD_AGENT_CLUSTER":"CUSR1C222","WVD_AGENT_RING":"R1","WVD_MONITORING_CONFIG_VERSION":"1.9999","WVD_MONITORING_DATA_DIRECTORY":"C:\\Windows\\system32\\config\\systemprofile\\AppData\\Roaming\\Microsoft\\Monitoring","WVD_MONITORING_ENV":"PROD","WVD_MONITORING_GCS_ACCOUNT":"RDSAgentProd","WVD_MONITORING_GCS_AUTH_ID":"RDSAgentPROD.GENEVA.KEYVAULT.RDSAGENT.WVD","WVD_MONITORING_GCS_AUTH_ID_TYPE":"AuthKeyVault","WVD_MONITORING_GCS_CERTSTORE":"LOCAL_MACHINE\\MY","WVD_MONITORING_GCS_ENVIRONMENT":"Diagnostics Prod","WVD_MONITORING_GCS_NAMESPACE":"RDSAgentProd","WVD_MONITORING_GCS_REGION":"westus2","WVD_MONITORING_MDM_ACCOUNT":"RDSAgentProd","WVD_MONITORING_RESOURCE":"6bdff336-9a7e-4dd2-a161-247921833a36","WVD_MONITORING_ROLE":"f330d97ff026408393884f97ea993f79","WVD_MONITORING_ROLE_INSTANCE":"CPC-vamyl-W9UOR","WVD_MONITORING_TENANT":"PROD","__COMPAT_LAYER":"DetectorsAppHealth","windir":"C:\\Windows"}
[cmake] CMake Error: The source directory "C:/Users/v-amyli3/Desktop/Bi-weekly-app/demo%20project" does not exist.
[cmake] Specify --help for usage, or press the help button on the CMake GUI.
[proc] The command: "C:\Program Files\Microsoft Visual Studio\2022\Enterprise\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin\cmake.exe" -DCMAKE_BUILD_TYPE=Debug "-DCMAKE_INSTALL_PREFIX=C:/Users/v-amyli3/Desktop/Bi-weekly-app/demo project/out/install/MyPresets" -S C:/Users/v-amyli3/Desktop/Bi-weekly-app/demo%20project -B C:/Users/v-amyli3/Desktop/Bi-weekly-app/demo%20project/out/build/MyPresets -G Ninja exited with code: 1
[extension] [1747] cmake.configure finished (returned 1)
```

### Additional Information

_No response_

--- a/src/drivers/cmakeFileApiDriver.ts
+++ b/src/drivers/cmakeFileApiDriver.ts
@@ -221,8 +221,8 @@ export class CMakeFileApiDriver extends CMakeDriver {
             }
         }
         // -S and -B were introduced in CMake 3.13 and this driver assumes CMake >= 3.15
-        args.push("-S", `${encodeURI(util.lightNormalizePath(this.sourceDir))}`);
-        args.push("-B", `${encodeURI(util.lightNormalizePath(binaryDir))}`);
+        args.push("-S", `${util.lightNormalizePath(this.sourceDir)}`);
+        args.push("-B", `${util.lightNormalizePath(binaryDir)}`);
 
         if (!has_gen) {
             const generator = (configurePreset) ? {

--- a/src/presets/preset.ts
+++ b/src/presets/preset.ts
@@ -2255,10 +2255,10 @@ export function configureArgs(preset: ConfigurePreset): string[] {
     }
 
     if (preset.toolchainFile) {
-        result.push(`-DCMAKE_TOOLCHAIN_FILE=${encodeURI(preset.toolchainFile)}`);
+        result.push(`-DCMAKE_TOOLCHAIN_FILE=${preset.toolchainFile}`);
     }
     if (preset.installDir) {
-        result.push(`-DCMAKE_INSTALL_PREFIX=${encodeURI(preset.installDir)}`);
+        result.push(`-DCMAKE_INSTALL_PREFIX=${preset.installDir}`);
     }
 
     // Warnings

